### ISSUE TITLE
ListItem button onPress slow performance 

### ISSUE THREAD

**Issue by obykoo** [GeekyAnts/NativeBase#662]
I am having performance issue with List and onPress. onPress is delayed too much. I do not know if there is problem on iOS , but there is on Android.

![2017-03-20 01h42_04](https://cloud.githubusercontent.com/assets/25873477/24086417/4e5921be-0d0f-11e7-93c9-f58518bbd1f7.gif)

**Comment by obykoo** [comment]
Interesting => this issue go away if I use **ListView** from React instead of **List** from NativeBase

**Comment by obykoo** [comment]
Ok I found out why this is happening.
If you wrap <List> in <Content></Content>, then onEndReached in List will keep firing to infinity......

**Comment by srhise** [comment]
Can you elaborate by what you mean 'wrap list in content'?

**Comment by sankhadeeproy007** [comment]
Do not use List inside Content

**Comment by srhise** [comment]
So, one of the reasons to use native base is the nice setup of <header> <content> <footer> where content scrolls and header and footer are pinned to the top and botttom, respecitively.. When I remove <content> from wrapping a list that breaks that layout. How should one address that pattern using a list, without using the <content> wrapper?

**Comment by maotora** [comment]
@srhise did you solve this without using <Content>..</Content> ?

I'm facing a similar situation and would like to get some feedback on possible solutions before I open another issue.

**Comment by SupriyaKalghatgi** [comment]
@cyclops24 Solution provided in above [comment](https://github.com/GeekyAnts/NativeBase/issues/662#issuecomment-287672302) dd'nt work?

**Comment by mnlbox** [comment]
I also find this: facebook/react-native#11809
Maybe it's related to React-Native version. Now NativeBase is using version 0.43.0 but it seems performance issue for list and button fixed in React-Native version 0.44.0.

**Comment by shivrajkumar** [comment]
@mnlbox Check http://docs.nativebase.io/docs/Compatibility.html
And also It is not recommended to use Content inside List.

**Comment by srhise** [comment]
Wait, is it list inside content or content inside list??

Can you explain why that is not recommended, ie. what is the impact?

**Comment by mnlbox** [comment]
@srhise I think you right.
This issues is about `List` inside `Content`.

**/CC:** @shivrajkumar

**Comment by mnlbox** [comment]
Guys I have same issue. Please help me to fix this issue. 😢 

**/CC:** @SupriyaKalghatgi , @obykoo

**Comment by SupriyaKalghatgi** [comment]
NativeBase List will be deprecated, very soon this will be documented
So please use React Native's FlatList in place of it
You can use ListItem for the necessary styling
[#759](https://github.com/GeekyAnts/NativeBase/issues/759)

**Comment by grzjos** [comment]
@SupriyaKalghatgi Any update on NativeBase List? Will it be replaced by something else or should we stick to FlatList? Thanks!

**Comment by 1c7** [comment]
getting same problem, `Button`(from NativeBase) in `FlatList`(from ReactNative) sometime could take 1 to 2 second to respond to `onPress`
still trying to solve the problem

using React Native 0.52, and develop for Android.

**Comment by SupriyaKalghatgi** [comment]
@grzjos Go on with FlatList

**Comment by Laurensdc** [comment]
@SupriyaKalghatgi Didn't help performance for me. It's like the TouchableOpacity blocks screen rendering.

**Comment by 1c7** [comment]
update: I think the problem is `FlatList` from React Native, not the `Button` by `NativeBase`
I replace `FlatList` with LargeList: https://github.com/bolan9999/react-native-largelist
is work for me, performance get a lot better,

no idea why a third party lib have better performance then FlatList, but it work for me.

**Comment by SupriyaKalghatgi** [comment]
@Laurensdc Are you using FlatList or NativeBase `List`?

**Comment by Laurensdc** [comment]
@SupriyaKalghatgi I used both and had the same problems. I did keep the <ListItem> component from NativeBase in both scenarios, but I tried wrapping it in FlatList as suggested. This did not help performance unfortunately.

What was interesting is that the actual dispatch of the onPress() comes late. It's like the screen goes into 'pressed' state for a whole second, and only then starts to dispatch the function. The Redux actions for example are only dispatched after that second of delay.
